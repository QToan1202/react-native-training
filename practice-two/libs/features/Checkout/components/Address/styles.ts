import { StyleSheet } from 'react-native'

import { COLORS, FONT_FAMILY } from '@practice-two/shared'

const styles = StyleSheet.create({
  container: {
    paddingVertical: 15,
    paddingHorizontal: 20,
    backgroundColor: COLORS.WHITE,
  },
  btn: {
    paddingHorizontal: 23,
    paddingVertical: 4,
  },
  btnTitle: {
    fontSize: 12,
  },
  text: {
    fontFamily: FONT_FAMILY.MONTSERRAT[500],
    color: COLORS.GRAY_50,
  },
  address: {
    opacity: 0.7,
  },
  spacing: {
    rowGap: 4,
  },
})

export default styles
