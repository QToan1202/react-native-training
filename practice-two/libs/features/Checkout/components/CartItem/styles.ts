import { GetProps, styled } from 'tamagui'
import { Image } from 'react-native'

import { Paragraph } from '@practice-two/shared'

export const StyledText = styled(Paragraph, {
  content: '',
  fontWeight: '$2',
  color: '$color.gray_50',
  textTransform: 'capitalize',
})

export type StyledTextProps = GetProps<typeof StyledText>

export const StyledImage = styled(Image, {
  source: { uri: '' },
  width: 102,
  height: 102,
  borderRadius: 10, // getTokenValue('$radius.5')
})

export type StyledImageProps = GetProps<typeof StyledImage>
