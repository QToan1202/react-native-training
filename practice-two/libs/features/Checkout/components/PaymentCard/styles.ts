import { GetProps, styled } from 'tamagui'
import { Image, ImageBackground, StyleSheet } from 'react-native'

import Paragraph from '@components/Paragraph'
import { COLORS } from '@constants'

export const StyledImageBackground = styled(ImageBackground, {
  name: 'ImageBackground',
  source: { uri: '' },
  flexDirection: 'row',
  alignItems: 'flex-end',
  alignSelf: 'flex-start',
  paddingVertical: '$space.3',
  paddingHorizontal: 17,
  columnGap: 17,
  borderRadius: 9,
  overflow: 'hidden',
  maxHeight: 146,
  maxWidth: 246,
})

export type StyledImageBackgroundProps = GetProps<typeof StyledImageBackground>

export const StyledTitle = styled(Paragraph, {
  content: '',
  color: '$color.gray_200',
  fontSize: 13,
  textTransform: 'capitalize',
})

export type StyledTitleProps = GetProps<typeof StyledTitle>

export const StyledImage = styled(Image, {
  name: 'IconImage',
  source: { uri: '' },
  position: 'absolute',
  zIndex: '$zIndex.1',
  right: '$space.-0.5',
  bottom: '$space.-2.5',
})

export type StyledImageProps = GetProps<typeof StyledImage>

export const styles = StyleSheet.create({
  border: {
    paddingVertical: 36,
    paddingHorizontal: 38,
    alignItems: 'center',
    borderWidth: 2, // '$space.1'
    borderRadius: 10, // '$radius.5'
    borderColor: COLORS.GRAY_300, // '$color.gray_300'
    borderStyle: 'dashed',
  },
})
