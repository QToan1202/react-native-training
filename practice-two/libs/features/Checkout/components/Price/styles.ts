import { StyleSheet } from 'react-native'

import { COLORS, FONT_FAMILY } from '@practice-two/shared/constants'

const styles = StyleSheet.create({
  text: {
    fontFamily: FONT_FAMILY.MONTSERRAT[500],
    lineHeight: 20,
    color: COLORS.BLACK,
    textTransform: 'capitalize',
  },
  title: {
    paddingVertical: 14,
    fontFamily: FONT_FAMILY.MONTSERRAT[600],
    fontSize: 18,
  },
})

export default styles
