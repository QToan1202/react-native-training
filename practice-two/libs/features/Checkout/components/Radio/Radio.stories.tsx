import { ComponentMeta, ComponentStory } from '@storybook/react-native'

import { PAYMENT_METHODS } from '@practice-two/shared'

import { Radio } from '../index'

export default {
  title: 'components/Radio',
  component: Radio,
} as ComponentMeta<typeof Radio>

const Template: ComponentStory<typeof Radio> = (args) => <Radio {...args} />

export const Default = Template.bind({})
Default.args = {
  radioList: PAYMENT_METHODS,
}
