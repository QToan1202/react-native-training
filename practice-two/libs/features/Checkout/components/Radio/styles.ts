import { StyleSheet } from 'react-native'
import { GetProps, RadioGroup, styled } from 'tamagui'

import { COLORS } from '@practice-two/shared'

const styles = StyleSheet.create({
  normal: {
    borderColor: COLORS.RADIO_NORMAL,
  },
  selected: {
    borderColor: COLORS.RADIO_ACTIVE,
  },
})

export const StyledItem = styled(RadioGroup.Item, {
  unstyled: true,
  value: '',
  id: undefined,
  justifyContent: 'center',
  alignItems: 'center',
  width: '$radio.frame',
  height: '$radio.frame',
  borderRadius: '$radius.12',
  borderWidth: '$space.0.5',
})

export type StyledItemProps = GetProps<typeof StyledItem>

export const StyledIndicator = styled(RadioGroup.Indicator, {
  unstyled: true,
  width: '$radio.indicator',
  height: '$radio.indicator',
  backgroundColor: '$color.radio_active',
  borderRadius: '$radius.12',
})

export type StyledIndicatorProps = GetProps<typeof StyledIndicator>

export default styles
