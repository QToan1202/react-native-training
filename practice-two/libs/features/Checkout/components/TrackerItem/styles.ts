import { StyleSheet } from 'react-native'

import { COLORS, FONT_FAMILY } from '@practice-two/shared'

const styles = StyleSheet.create({
  container: {
    rowGap: 6,
  },
  text: {
    fontFamily: FONT_FAMILY.MONTSERRAT[500],
    color: COLORS.GRAY_400,
    fontSize: 10,
    lineHeight: 18,
  },
  title: {
    color: COLORS.DARK_50,
    fontSize: 14,
    textTransform: 'capitalize',
  },
})

export default styles
