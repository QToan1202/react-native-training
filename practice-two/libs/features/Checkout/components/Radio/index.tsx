import { memo, useCallback, useMemo, useState } from 'react'
import isEqual from 'react-fast-compare'
import { Label, RadioGroup, RadioGroupProps, Separator, XStack, YStack } from 'tamagui'

import { IRadioItem } from '@types'

import styles, { StyledIndicator, StyledItem } from './styles'

export type RadioProps = RadioGroupProps & {
  radioList: IRadioItem[]
  onChangeValue?: (value: string) => void
}

const Radio = ({ radioList, onChangeValue, ...rest }: RadioProps) => {
  const [selectedItem, setSelectedItem] = useState<string>()
  const handleSelectedRadio = useCallback(
    (value: string): void => {
      if (onChangeValue) onChangeValue(value)
      setSelectedItem(value)
    },
    [onChangeValue]
  )
  const renderList = useMemo(() => {
    return radioList.map(({ id, name }: IRadioItem, index: number) => (
      <YStack key={id}>
        <XStack
          alignItems="center"
          space="$space.2"
          paddingLeft="$space.3.5"
          paddingVertical="$space.3"
        >
          <StyledItem
            value={name}
            id={id}
            style={selectedItem === name ? styles.selected : styles.normal}
          >
            <StyledIndicator />
          </StyledItem>

          <Label unstyled htmlFor={id} fontWeight="$3" color="$color.gray_50" lineHeight="$4">
            {name}
          </Label>
        </XStack>
        {/* Do not render Separator component on the last item */}
        {index !== radioList.length - 1 && (
          <Separator alignSelf="stretch" borderColor="$color.divider" />
        )}
      </YStack>
    ))
  }, [radioList, selectedItem])

  return (
    <RadioGroup onValueChange={handleSelectedRadio} {...rest}>
      <YStack>{renderList}</YStack>
    </RadioGroup>
  )
}

export default memo(Radio, isEqual)
