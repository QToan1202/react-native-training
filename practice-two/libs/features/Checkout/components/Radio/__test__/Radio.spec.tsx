import { fireEvent, render, screen } from '@testing-library/react-native'

import { PAYMENT_METHODS } from '@constants'

import Radio from '../index'
import styles from '../styles'

describe('Testing Radio component', () => {
  beforeEach(() => {
    render(<Radio radioList={PAYMENT_METHODS} testID="radio-item" />)
  })

  it('Render Radio component success', () => {
    const component = render(<Radio radioList={PAYMENT_METHODS} />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('Radio title will not change style when press item', () => {
    const component = screen.getByText('Wallet')

    fireEvent.press(component)
    expect(component).toHaveStyle(styles.title)
  })
})
