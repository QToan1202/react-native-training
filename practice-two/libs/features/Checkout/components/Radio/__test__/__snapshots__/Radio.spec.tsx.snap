// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Radio component Render Radio component success 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    [
      {
        "id": "1",
        "name": "Debit / Credit Card",
      },
      {
        "id": "2",
        "name": "Netbanking",
      },
      {
        "id": "3",
        "name": "Cash on Delivery",
      },
      {
        "id": "4",
        "name": "Wallet",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingLeft": 16,
            "paddingVertical": 13.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "columnGap": 4,
                "flexDirection": "row",
              },
              {
                "columnGap": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "height": 18,
                  "justifyContent": "center",
                  "width": 18,
                },
                {
                  "borderColor": "#ced2d9",
                },
              ]
            }
          />
          <Text
            style={
              [
                {
                  "color": "#fff",
                  "fontFamily": "Montserrat_400Regular",
                },
                {
                  "fontSize": 14,
                },
                {
                  "color": "#4f4f4f",
                  "fontFamily": "Montserrat_600SemiBold",
                  "lineHeight": 21,
                },
              ]
            }
          >
            Debit / Credit Card
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#d8d8d8",
            "height": 1,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingLeft": 16,
            "paddingVertical": 13.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "columnGap": 4,
                "flexDirection": "row",
              },
              {
                "columnGap": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "height": 18,
                  "justifyContent": "center",
                  "width": 18,
                },
                {
                  "borderColor": "#ced2d9",
                },
              ]
            }
          />
          <Text
            style={
              [
                {
                  "color": "#fff",
                  "fontFamily": "Montserrat_400Regular",
                },
                {
                  "fontSize": 14,
                },
                {
                  "color": "#4f4f4f",
                  "fontFamily": "Montserrat_600SemiBold",
                  "lineHeight": 21,
                },
              ]
            }
          >
            Netbanking
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#d8d8d8",
            "height": 1,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingLeft": 16,
            "paddingVertical": 13.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "columnGap": 4,
                "flexDirection": "row",
              },
              {
                "columnGap": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "height": 18,
                  "justifyContent": "center",
                  "width": 18,
                },
                {
                  "borderColor": "#ced2d9",
                },
              ]
            }
          />
          <Text
            style={
              [
                {
                  "color": "#fff",
                  "fontFamily": "Montserrat_400Regular",
                },
                {
                  "fontSize": 14,
                },
                {
                  "color": "#4f4f4f",
                  "fontFamily": "Montserrat_600SemiBold",
                  "lineHeight": 21,
                },
              ]
            }
          >
            Cash on Delivery
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#d8d8d8",
            "height": 1,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingLeft": 16,
            "paddingVertical": 13.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "columnGap": 4,
                "flexDirection": "row",
              },
              {
                "columnGap": 8,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "height": 18,
                  "justifyContent": "center",
                  "width": 18,
                },
                {
                  "borderColor": "#ced2d9",
                },
              ]
            }
          />
          <Text
            style={
              [
                {
                  "color": "#fff",
                  "fontFamily": "Montserrat_400Regular",
                },
                {
                  "fontSize": 14,
                },
                {
                  "color": "#4f4f4f",
                  "fontFamily": "Montserrat_600SemiBold",
                  "lineHeight": 21,
                },
              ]
            }
          >
            Wallet
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
